src/connection-manager/chains.ts

```diff
@@ -31,6 +33,39 @@ function addLedgerToChain(ledger: Ledger, chain: Chain): void {
   chain.updated = ledger.first_seen
 }
 
+/**
+ * Saves the chain id for each validator known to be in a given chain.
+ *
+ * @param chain - A chain object.
+ * @returns Void.
+ */
+async function saveValidatorChains(chain: Chain): Promise<void> {
+  let id = chain.id
+  const lists = await getLists().catch((err) => {
+    log.error('Error getting validator lists', err)
+    return undefined
+  })
+  if (lists != null) {
+    Object.entries(lists).forEach(([network, set]) => {
+      if (overlaps(chain.validators, set)) {
+        id = network
```

Comment:
Is this guaranteed to only happen once? (Or can there be multiple matches? - In which case do we always want the last one on the list?)

Reply:
Rewriting this code is way out of the scope of this PR. I just had to move the function to avoid a circular dependency.

Reply:
Ah I see, I didn't realize it was just moved.